projects/standard-rulesets/src/lintgpt/index.ts

```diff
@@ -0,0 +1,307 @@
+  ) {
+    super();
+  }
+
+  async runRulesV2(inputs: {
+    context: any;
+    diffs: ObjectDiff[];
+    fromSpec: OpenAPIV3.Document;
+    toSpec: OpenAPIV3.Document;
+    groupedFacts: OpenAPIFactNodes;
+  }): Promise<RuleResult[]> {
+    const operationsToRun: AIRuleRunInputs[] = [];
+    const responsesToRun: AIRuleRunInputs[] = [];
+    const propertiesToRun: AIRuleRunInputs[] = [];
+
+    const lintgptRulesHelper = new LintgptRulesHelper(this.lintgptClient);
+
+    inputs.groupedFacts.endpoints.forEach((endpoint) => {
+      const { path, method } = endpoint;
+
+      const location = `${method} ${path}`;
+      const didChange = endpoint.change?.changeType === ChangeType.Changed;
+      const jsonPath = (endpoint.after?.location.jsonPath ||
+        endpoint.before?.location.jsonPath)!;
+
+      const inAfterSpec = endpoint.after !== null;
+      if (inAfterSpec) {
+        operationsToRun.push({
+          locationContext: location,
+          jsonPath,
+          value: jsonPointerHelpers.get(inputs.toSpec, jsonPath),
+          before: didChange
+            ? jsonPointerHelpers.get(
+                inputs.fromSpec,
+                endpoint.before?.location.jsonPath!
+              )
+            : undefined,
+        });
+      }
+
+      endpoint.responses.forEach((responses) => {
+        const location = `${method} ${path} ${responses.statusCode} response`;
+        const didChange = responses.change?.changeType === ChangeType.Changed;
+
+        responses.bodies.forEach((body) => {
+          body.fields.forEach((property) => {
+            if (property.after) {
+              const propertyLocation = `${method} ${path} ${responses.statusCode} response property. Name: \`${property.after.value.key}\`. Required? \`${property.after.value.required}\``;
+              const didChange =
+                property.change?.changeType === ChangeType.Changed;
+
+              propertiesToRun.push({
+                locationContext: propertyLocation,
+                jsonPath: property.after.location.jsonPath,
+                value: jsonPointerHelpers.get(
+                  inputs.toSpec,
+                  property.after.location.jsonPath
+                ),
+                before: didChange
+                  ? jsonPointerHelpers.get(
+                      inputs.fromSpec,
+                      property.before!.location.jsonPath
+                    )
+                  : undefined,
+              });
+            }
+          });
+        });
+
+        responsesToRun.push({
+          locationContext: location,
+          jsonPath,
+          value: jsonPointerHelpers.get(inputs.toSpec, jsonPath),
+          before: didChange
+            ? jsonPointerHelpers.get(
+                inputs.fromSpec,
+                responses.before?.location.jsonPath!
+              )
+            : undefined,
+        });
+      });
+    });
+
+    ///// run the rules
+
+    const operationsRules = this.requirementRules.filter(
```

Comment:
@Alexandre-Herve -- does this support parameters / headers? Is that coming next. Just want to make sure I write down the limitations accurately in the docs. I know it's just busy work to iterate those so we can release it after v1 if we are ready to get the bulk of this out.

Reply:
I just copied what you had done there, but this can be added later yes